templates/migration-guide-section.html

```diff
@@ -0,0 +1,101 @@
+{% extends "layouts/page-with-menu.html" %}
+{% import "macros/book.html" as book_macros %}
+
+{% block head_extensions %}
+    <script src="/optional-helpers.js"></script>
+{% endblock %}
+
+{% block page_name %}Migration Guide{% endblock %}
```

Comment:
<img width="1944" alt="image" src="https://user-images.githubusercontent.com/188612/224673403-c5a40459-b6da-43f7-9807-a8d37aac0c92.png">

I think this can be moved to a new layout: 
- create a `layouts/docs.html` (or whatever name fits better)
- make both `templates/book-section.html` and `templates/migration-guide-section.html` extend the new layout
- To pass the `path` var you can do: https://github.com/Keats/tera/issues/692#issuecomment-977755788
- I think `page_name` can be set by a "gran-children" 🤞 Otherwise, you could add a `docs_title` var and pass it with the path.

It would look something like:
```html
{% extends "layouts/docs.html" %}

{% block page_name %}Book{% endblock %}

{% block docs_vars %}
    {% set docs_root_path = "learn/book/_index.md" %}
{% endblock %}
```

Reply:
Beautiful, thanks @doup. I knew there was a lot of shared functionality but I was having a hard time understanding the zola functions. I'll take a look at incorporating your changes.

Reply:
@doup Unfortunately, I don't _think_ this route works (unless I'm misunderstanding, which may very well be the case). This is what I have in my `layout/templates/docs.html`:
```
...
{% block page_name %}Docs{% endblock %}

{% block docs_vars %}
    {% set docs_root_path = "" %}
{% endblock %}

{% block mobile_page_menu %}
    {{book_macros::book_menu(prefix="mobile-menu", root=get_section(path=docs_root_path))}}
{% endblock %}

{% block page_menu %}
    {{book_macros::book_menu(prefix="page-menu", root=get_section(path=docs_root_path))}}
{% endblock %}

...
```

But it gives:
```bash
Reloading only template
Error: Failed to build the site
Error: Failed to render section '/Users/kap/dev/bevy-website/content/learn/book/getting-started/setup/_index.md'
Error: Reason: Failed to render 'book-section.html' (error happened in a parent template)
Error: Reason: Function call 'get_section' failed
Error: Reason: Variable `docs_root_path` not found in context while rendering 'book-section.html'
Done in 1.0s.
```
Which I guess makes sense because the `docs_root_path` is in a different "block context". I've tried `path=docs_vars.docs_root_path`, but no luck. I've also tried `path={{docs_root_path}}`, which gives a parsing error. `set_global` also doesn't work. I'm not sure how to share variables between blocks and the docs don't showcase that feature. I'm still trying to wrap my head around the templating language and what's available, but right now I'm having no luck.

Reply:
Here: https://github.com/akappel/bevy-website/pull/1 

The thing is that as the error says the block must be defined before the vars are being used; the solution in this case was to move the block definition to the top of the root layout (`base.html`). This way it works.

I've changed the block name to `extra_data` as suggested in the linked issue, this way is more generic and makes sense for any layout that extends `base`.